homeassistant/components/ads/cover.py

```diff
@@ -0,0 +1,175 @@
+"""Support for ADS covers."""
+import logging
+
+import voluptuous as vol
+
+from homeassistant.components.cover import (
+    PLATFORM_SCHEMA, SUPPORT_OPEN, SUPPORT_CLOSE, SUPPORT_STOP,
+    SUPPORT_SET_POSITION, ATTR_POSITION, DEVICE_CLASSES_SCHEMA,
+    CoverDevice)
+from homeassistant.const import (
+    CONF_NAME, STATE_OPEN, STATE_CLOSED, CONF_DEVICE_CLASS)
+import homeassistant.helpers.config_validation as cv
+
+from . import CONF_ADS_VAR, CONF_ADS_VAR_POSITION, DATA_ADS, \
+    AdsEntity, STATE_KEY_STATE, STATE_KEY_POSITION
+
+_LOGGER = logging.getLogger(__name__)
+
+DEFAULT_NAME = 'ADS Cover'
+
+CONF_ADS_VAR_SET_POS = 'adsvar_set_position'
+CONF_ADS_VAR_OPEN = 'adsvar_open'
+CONF_ADS_VAR_CLOSE = 'adsvar_close'
+CONF_ADS_VAR_STOP = 'adsvar_stop'
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Optional(CONF_ADS_VAR): cv.string,
+    vol.Optional(CONF_ADS_VAR_POSITION): cv.string,
+    vol.Optional(CONF_ADS_VAR_SET_POS): cv.string,
+    vol.Optional(CONF_ADS_VAR_CLOSE): cv.string,
+    vol.Optional(CONF_ADS_VAR_OPEN): cv.string,
+    vol.Optional(CONF_ADS_VAR_STOP): cv.string,
+    vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+    vol.Optional(CONF_DEVICE_CLASS): DEVICE_CLASSES_SCHEMA
+})
+
+
+def setup_platform(hass, config, add_entities, discovery_info=None):
+    """Set up the cover platform for ADS."""
+    ads_hub = hass.data.get(DATA_ADS)
+
+    ads_var_is_closed = config.get(CONF_ADS_VAR)
+    ads_var_position = config.get(CONF_ADS_VAR_POSITION)
+    ads_var_pos_set = config.get(CONF_ADS_VAR_SET_POS)
+    ads_var_open = config.get(CONF_ADS_VAR_OPEN)
+    ads_var_close = config.get(CONF_ADS_VAR_CLOSE)
+    ads_var_stop = config.get(CONF_ADS_VAR_STOP)
+    name = config.get(CONF_NAME)
```

Comment:
Use `dict[key]` for required config keys and keys with default config schema values.